pkg/dockerfile/generator_test.go

```diff
@@ -46,29 +50,54 @@ RUN curl https://pyenv.run | bash && \
 `, version, version)
 }
 
-func TestGenerateEmptyCPU(t *testing.T) {
-	tmpDir, err := os.MkdirTemp("", "test")
-	require.NoError(t, err)
+func testPreamble() string {
+	// Get default environment variables: the hardcoded parts. (Excluding override-able parts.)
+	return `ENV DEBIAN_FRONTEND=noninteractive
+ENV PYTHONUNBUFFERED=1
+ENV LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib/x86_64-linux-gnu:/usr/local/nvidia/lib64:/usr/local/nvidia/bin
+`
+}
 
-	conf, err := config.FromYAML([]byte(`
-build:
-  gpu: false
-predict: predict.py:Predictor
-`))
-	require.NoError(t, err)
-	require.NoError(t, conf.ValidateAndCompleteConfig())
+func testPreambleDefault() string {
+	// Get default environment variables INCLUDING the override-able parts (like cache home)
+	return testPreamble() + `ENV XDG_CACHE_HOME=/src/.cache
+`
+}
 
+func testGenerate(t *testing.T, conf *config.Config) (*Generator, string) {
+	tmpDir, err2 := os.MkdirTemp("", "test")
+	require.NoError(t, err2)
 	gen, err := NewGenerator(conf, tmpDir)
 	require.NoError(t, err)
 	actual, err := gen.Generate()
 	require.NoError(t, err)
+	return gen, actual
+}
+
+func testConfigYaml(t *testing.T, cogYaml string) *config.Config {
+	conf, err := config.FromYAML([]byte(cogYaml))
+	require.NoError(t, err)
+	require.NoError(t, conf.ValidateAndCompleteConfig())
+	return conf
+}
+
+//===============================================================================
+// tests: build.gpu
+//-------------------------------------------------------------------------------
+
```

Comment:
ℹ️  This diff is hard to read in "unified" diff, I'd def recommend using side-by-side or viewing the file.